rm/rm.h

```diff
@@ -1951,10 +1955,14 @@ M0_INTERNAL void m0_rm_remote_fini(struct m0_rm_remote *rem);
 
 #define M0_RM_REMOTE_PUT(rem) \
 { \
-	int cnt = m0_ref_read(&(rem)->rem_refcnt); \
+	int cnt; \
+	struct m0_mutex *mux = &rem->rem_resource->r_mutex; \
+	m0_mutex_lock(mux); \
+	cnt = m0_ref_read(&(rem)->rem_refcnt); \
```

Comment:
Store "rem" in a local variable to avoid multiple evaluations. Wrap the macro body in ({ ... }) — it's an expression, not statement.

Also, now that ->rem_refcnt is explicitly protected by a mutex, why do you need an atomic counter? Let's just use an integer with M0_CNT_{INC,DEC}().

Reply:
It's explicitly protected by a mutex only on decrements. The increments still don't need any protection.